src/camvidlog2/yoloe/ai.py

```
@@ -147,54 +153,70 @@ def generate_tracked_bboxes(
-        if len(results_filtered) == 0:
-            yield pd.DataFrame(columns=columns)
-            continue
-
-        # Do tracking
+        all_detections = []
+
+        # beware, batch size is baked into the model too
+        for batch in itertools.batched(
+            slice_frame(frame, slice_width=640, slice_height=640, slice_overlap=0.25),
+            max_batch_size,
+        ):
```

Ruff (0.11.9)

`itertools.batched()` without an explicit `strict` parameter

(B911)